main/src/cgeo/geocaching/connector/lc/LCApi.java

```diff
@@ -187,10 +191,12 @@ private static Geocache parseCacheDetail(final JsonNode response) {
             final String[] segments = firebaseDynamicLink.split("/");
             final String id = segments[segments.length - 1];
             final String uuid = response.get("Id").asText();
+            final Boolean isComplete = response.get("IsComplete").asBoolean();
+            final String geocode = "LC" + uuid;
             final String ilink = response.get("KeyImageUrl").asText();
             final String desc = response.get("Description").asText();
             cache.setReliableLatLon(true);
-            cache.setGeocode("LC" + uuid);
+            cache.setGeocode(geocode);
```

Comment:
I have tried for a long time to avoid referencing "LC" hardcoded here. There is a getname() in the connector but no matter how I tried to referene it via getname() the compile always failed with something like can't use static on a non static or so. This is clearly due to my lack of Java skills at the moment.

Reply:
You can create a private (or public?) class variable (similar to like there already is with `CACHE_URL` in LCConnector) which then can be used in `getNameAbbreviated()` as well as the LCApi...

Reply:
`getNameAbbreviated()` is already there in LCconnector. What I don't know is how to reference this from within LCApi. Whatever I tried did not compile.

Reply:
Yes, it is only available if you have an instance of LCConnector. Therefore, you need to create a class variable in `LCConnector` like:

```
    @NonNull
    protected static final String GEOCODE_PREFIX = "LC";
```
(`protected` means, it is only available in the current package. In your case `cgeo.geocaching.connector.lc;`)

And then you can use it everywhere in the code:

e.g. `cache.setGeocode(LCConnector.GEOCODE_PREFIX + uuid);` or 

```
    @Override
    @NonNull
    public String getNameAbbreviated() {
        return GEOCODE_PREFIX;
    }
```


---

But I think this is definitely optional. Do it the way you like it, for me the current code would be fine too.

Reply:
I am going to fix it. My experience is such that I will always go back to those code sequences I once understood and use those mechanisms then in other places. This is now such a code sequence.